packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -31,24 +35,77 @@ const propTypes = {
 const defaultProps = {
   disableUnselectedRows: false,
   onChange: undefined,
+  onBlur: undefined,
   selectedIndexes: [],
 };
 
+const rowsSelector = 'row-group';
+
 class SelectableTableRows extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+
+  componentDidMount() {
+    // Adds ARIA-Label for Pre-selected rows
+    const rowGroup = document.querySelector(`#${rowsSelector}`);
+    for (let rowIndex = 0; rowIndex < rowGroup.children.length; rowIndex += 1) {
+      const preselectedRow = rowGroup.children[rowIndex];
+      let ariaLabel = 'pre-selected row';
+      if ((preselectedRow.hasAttribute('aria-selected')) && (preselectedRow.getAttributeNode('aria-selected').value === 'true')) {
+        for (let columnIndex = 0; columnIndex < preselectedRow.children.length; columnIndex += 1) {
+          ariaLabel += ` ${preselectedRow.children[columnIndex].innerText}`;
+        }
+        rowGroup.children[rowIndex].setAttribute('aria-label', ariaLabel);
+      }
+    }
```

Comment:
You can have this logic in a separate sub method and call that method from here.

Reply:
having methods without `this` : throws lint error : class-methods-use-this. hence having this line of code inside componentDidMount() method and only static class methods inside class can be declared without `this` .